src/webapp/oasis_managers.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import ldap
+
+
+def get_oasis_manager_endpoint_info(global_data, vo, ldappass):
+    """ return list of oasis manager info for endpoint with the structure:
+
+        [ {'ContactID': ContactID, 'Name': Name, 'DNs': DNs,
+           'CILogonID': CILogonID, 'sshPublicKeys': [sshPubKey, ...]}, ... ]
+
+        with one list entry per oasis manager for the given vo, whose
+        contact db info specifies a CILogonID, and whose cilogon info
+        for that CILogonID contains a list of ssh public keys. """
+        
+    managers = get_vo_oasis_managers(global_data, vo)
+    if not isinstance(managers, list) or if not managers:
+        return []
+    cilogon_id_map = get_cilogon_ldap_id_map(ldappass)
+    ssh_keys_map = cilogon_id_map_to_ssh_keys(cilogon_id_map)
+    contact_cilogon_ids = get_contact_cilogon_id_map(global_data)
+    info = []
+    for manager in managers:
+        ContactID = safe_dict_get(manager, 'ID')
+        if ContactID in contact_cilogon_ids:
+            Name = manager.get('Name')
+            DNs = manager.get('DNs', [])
+            CILogonID = contact_cilogon_ids[ContactID].cilogon_id
+            ssh_keys = ssh_keys_map.get(CILogonID, [])
+            info.append({
+                'ContactID': ContactID,
+                'Name': Name,
+                'DNs': DNs,
+                'CILogonID': CILogonID,
+                'sshPublicKeys': ssh_keys
+            })
+    return info
+
+
+def get_contact_cilogon_id_map(global_data):
+    """ return contacts dict, limited to users with a CILogonID """
+    contacts = global_data.get_contacts_data().users_by_id;
+    return { k: v for k, v in contacts.items() if v.cilogon_id is not None }
+
+
+def get_vo_oasis_managers(global_data, vo):
+    """return OASIS Managers dict for given vo, if any, else an empty dict"""
+    vos_data = global_data.get_vos_data()
+    return safe_dict_get(vos_data, vo, "OASIS", "Managers", default=[])
```

Comment:
```suggestion
    return safe_dict_get(vos_data, vo, "OASIS", "Managers", default={})
```

Reply:
Actually in 21526c5b96 i changed this from `{}` _to_ `[]`, because i think OASIS Managers is supposed to be a list.

Reply:
You should update the docstring to say list instead of dict, then.

Reply:
Good point!  Updated.